python-client/giskard/models/base/model.py

```diff
@@ -354,9 +354,11 @@ def upload(self, client: GiskardClient, project_key, validate_ds=None) -> str:
             This method saves the model to a temporary directory before uploading it. The temporary directory
             is deleted after the upload is completed.
         """
-        from giskard.core.model_validation import validate_model
+        from giskard.core.model_validation import validate_model, validate_model_loading_and_saving
+
+        reloaded_model = validate_model_loading_and_saving(self)
+        validate_model(model=reloaded_model, validate_ds=validate_ds)
```

Comment:
For us it's important to know that the deserialized model works. So I think we should validate the loaded one and if it fails we could also run validation on the original model to give a hint that it might be a problem with the serialization

Reply:
in fact the validation of the original model already happens in the scan, but it doesn't hurt running it in `upload()` too (just in case a user bypassed the scan).

So what you're suggesting @andreybavt is something like:
```py
reloaded_model = validate_model_loading_and_saving(self)
try:
  validate_model(model=reloaded_model, validate_ds=validate_ds)
except Exception as e_reloaded:
  try:
    validate_model(model=self, validate_ds=validate_ds)
  except Exception as e_loaded:
    raise e_loaded
  raise GiskardException("An error occured during the deserialisation of your model, please report this issue to giskard, etc.") from e_reloaded


```

Reply:
almost, we just need to make sure we don't lose the `e_reloaded` when you throw `e_loaded`

so something like
```
reloaded_model = validate_model_loading_and_saving(self)
try:
    validate_model(model=reloaded_model, validate_ds=validate_ds)
except Exception as e_reloaded:
    try:
        validate_model(model=self, validate_ds=validate_ds)
        logger.info("Original model validated successfully")
    except Exception as e_loaded:
        logger.exception("Failed to validate on original model", e)
    raise GiskardException(
        "An error occured while validating a deserialized version your model, please report this issue to Giskard") from e_reloaded

```

Reply:
ok done